packages/core/src/providers/keyp.ts

```diff
@@ -0,0 +1,142 @@
+ *
+ * By default, Auth.js assumes that the Keyp provider is
+ * based on the [OAuth 2](https://www.rfc-editor.org/rfc/rfc6749.html) specification.
+ *
+ * :::tip
+ *
+ * The Keyp provider comes with a [default configuration](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/keyp.ts).
+ * To override the defaults for your use case, check out [customizing a built-in OAuth provider](https://authjs.dev/guides/providers/custom-provider#override-default-options).
+ *
+ * :::
+ *
+ * :::tip
+ * If using JWT for the session, you can add additional properties to the JWT and session. Example:
+ * ```js
+ * options: {
+ *   callbacks: {
+ *      async jwt({ token, account, profile }: { token: Token; account: Account; profile: Profile }) {
+ *          if (account) {
+ *              // Comes from the returned JWT from Keyp
+ *              token.accessToken = account.access_token;
+ *          }
+ *          if (profile) {
+                // Comes from the /userinfo endpoint
+ *              token.username = profile.username;
+ *              token.address = profile.address;
+ *              token.sub = profile.sub;
+ *          }
+ *          return token;
+ *      },
+ *      async session({ session, token }: { session: Session; token: Token }) {
+ *          // Send properties to the client, like an access_token from a provider.
+ *          if (token) {
+ *              session.user.accessToken = token.accessToken;
+ *              session.user.username = token.username;
+ *              session.user.address = token.address;
+ *              session.user.id = token.sub;
+ *          }
+ *          return session;
+ *      },
+ *  },
+ * ```
+ * :::
+ *
+ * :::info **Disclaimer**
+ *
+ * If you think you found a bug in the default configuration, you can [open an issue](https://authjs.dev/new/provider-issue).
+ *
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ *
+ * :::
+ */
+export default function Keyp<P extends KeypProfile>(
+  config: OIDCUserConfig<P> & AdditionalConfig
+): OIDCConfig<P> {
+  const { clientId, scope } = config
+  const defaultScope = "openid email"
+
+  // Keyp's API domain is configurable for local testing, but defaults to https://api.usekeyp.com
+  const KEYP_API_DOMAIN =
+    process.env.NEXT_PUBLIC_KEYP_API_DOMAIN || "https://api.usekeyp.com"
+
+  return {
+    id: "keyp",
+    name: "Keyp",
+    type: "oidc",
+    clientId,
+    issuer: "https://api.usekeyp.com",
+    wellKnown: `${KEYP_API_DOMAIN}/oauth/.well-known/openid-configuration`,
+    authorization: { url: "https://app.usekeyp.com/oauth/auth", params: { scope: scope ? scope : defaultScope } },
+    token: { url: "https://app.usekeyp.com/oauth/token" },
```

Comment:
token endpoint should be on api subdomain. Only the authorization endpoint is through the app subdomain. Is there anywhere else this is incorrect?

https://api.usekeyp.com/oauth/.well-known/openid-configuration shows it correct `"https://api.usekeyp.com/oauth/token"`